### PR TITLE
opencv/4.8.1: fix access to wayland-protocols root path with 1 profile

### PR DESCRIPTION
**Pull request by SpaceIm**
Current workaround for `wayland-protocols` with 1 profile won't work anymore after https://github.com/conan-io/conan-center-index/pull/21239. It was working incidentally because `xkbcommon` was dragging `wayland-protocols` in requirements if 1 profile. But after https://github.com/conan-io/conan-center-index/pull/21239, `wayland-protocols` won't be visible in requirements anymore, and `self.dependencies` doesn't contain build requirements if 1 profile. Fortunately, legacy `self.deps_cpp_info` contains build requirements if 1 profile (and we can use `deps_cpp_info` in this branch, because 1 profile means that we are using conan v1 for sure).

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by EstebanDugueperoux2 — APPROVED** [review]

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -1550,7 +1534,23 @@ def generate(self):
...
         if self.options.get_safe("with_wayland"):
             deps = PkgConfigDeps(self)
-            if not self._is_legacy_one_profile:
+            if self._is_legacy_one_profile:
+                # Manually generate pkgconfig file of wayland-protocols since
+                # PkgConfigDeps.build_context_activated can't work with legacy 1 profile
+                # We must use legacy conan v1 deps_cpp_info because self.dependencies doesn't
+                # contain build requirements when using 1 profile.
+                wp_prefix = self.deps_cpp_info["wayland-protocols"].rootpath
+                wp_version = self.deps_cpp_info["wayland-protocols"].version
```
Comment (uilianries) [inline recipes/opencv/4.x/conanfile.py:1543]:
```suggestion
                wp_prefix = self.dependencies.build["wayland-protocols"].package_folder
                wp_version = self.dependencies.build["wayland-protocols"].ref.version
```

Dependencies.build is retro-compatible with Conan 1.x and will list wayland-protocol for sure.

Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py:1543]: It's not retro compatible with conan v1 but not with 1 profile.

Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py:1543]: So with 1 profile, `self.dependencies["wayland-protocols"]` is empty but not `self.dependencies.build["wayland-protocols"]`?

Comment (uilianries) [reply recipes/opencv/4.x/conanfile.py:1543]: Tool requires will be always available under `self.dependencies.build`, but not in `self.dependencies` directly. The number of profiles does not matter and the behavior is the same for Conan 1.x and 2.x

Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py]: ok, fixed by https://github.com/conan-io/conan-center-index/pull/21242/commits/9aad166fdedb76bd890662c5798d803b1a48703b

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by EstebanDugueperoux2** [comment]
@franramirez688 @uilianries @RubenRBS is this MR ready for merge for you?
Regards.

**Review by RubenRBS — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**